src/library/scala/collection/mutable/WrappedArray.scala

```diff
@@ -178,6 +178,30 @@ object WrappedArray {
       case that: ofInt => Arrays.equals(array, that.array)
       case _ => super.equals(that)
     }
+    override def sum[B >: Int](implicit num: Numeric[B]): B = if (num eq scala.math.Numeric.IntIsIntegral) {
+      var z = 0
+      var i = 0
+      val as = array
+      while (i < as.length) {
+        z += as(i)
+        i += 1
+      }
+      z
+    } else {
+      super.sum[B]
+    }
+    override def product[B >: Int](implicit num: Numeric[B]): B = if (num eq scala.math.Numeric.IntIsIntegral) {
+      var z = 0
+      var i = 0
+      val as = array
+      while (i < as.length) {
+        z *= as(i)
+        i += 1
+      }
+      z
+    } else {
+      super.sum[B]
```

Comment:
I think this should be `super.sum(num)`
avoid any implicit that may be in scope, and avoid the implicit scan
and for me  - makes it more obvious